server.py

```diff
@@ -39,6 +43,8 @@ def __init__(self, motd_content: Optional[Dict[str, List[str]]]) -> None:
         self.lock = threading.Lock()
         self.channels: Dict[str, Channel] = {}
         self.connected_users: Dict[str, UserConnection] = {}
+        self.host = host
+        self.port = str(port)
```

Comment:
Ports are numbers, so storing  them as strings doesn't make sense to me.

```suggestion
        self.port = port
```